src/classes/RD2_PauseForm_CTRL.cls

```diff
@@ -0,0 +1,270 @@
+*/
+public with sharing class RD2_PauseForm_CTRL {
+
+    /***
+    * @description Recurring Donation Schedule service
+    */
+    @TestVisible
+    private static RD2_ScheduleService.PauseScheduleHandler pauseHandler {
+        get {
+            if (pauseHandler == null) {
+                pauseHandler = new RD2_ScheduleService.PauseScheduleHandler();
+            }
+            return pauseHandler;
+        } set;
+    }
+
+    /**
+     * @description 
+     * @return 
+     */
+    @AuraEnabled(cacheable=false)
+    public static String getPauseData(Id rdId) {
+        String jsonData;
+        try {
+            PauseData pause = new PauseData();
+            pause.isRDClosed = isClosed(rdId);
+
+            if (!pause.isRDClosed) {
+                pause.pausedReason = buildPausedReason();
+                populateScheduleData(pause, rdId);
+            }
+
+            jsonData = JSON.serialize(pause);
+
+        } catch (Exception e) {
+            throwAuraHandledException(e.getMessage());
+        }
+
+        return jsonData;
+    }
+
+    /**
+     * @description 
+     * @return 
+     */
+    public static Boolean isClosed(Id rdId) {
+        Boolean isClosed = false;
+        try {
+            npe03__Recurring_Donation__c rd = [
+                SELECT Status__c
+                FROM npe03__Recurring_Donation__c
+                WHERE Id = :rdId
+                LIMIT 1
+            ];
+
+            isClosed = (new RD2_RecurringDonation(rd)).isClosed();
+
+        } catch (Exception e) {
+            //the record has been deleted; 
+        }
+
+        return isClosed;        
+    }
+
+    /**
+     * @description Returns the Paused Reason value and label select options
+     * @return PausedReason
+     */
+    private static PausedReason buildPausedReason() {
+        PausedReason reason = new PausedReason();
+        reason.label = System.Label.RD2_PausedReasonLabel;
+        reason.options = getPausedReasonOptions();
+        reason.placeholder = System.Label.lblSelectOptionPlaceholder;
+
+        return reason;
+    }
+
+    /**
+     * @description Returns the Paused Reason value and label select options
+     * @return void
+     */
+    private static void populateScheduleData(PauseData pause, Id rdId) {
+        RecurringDonationSchedule__c schedule = pauseHandler.getPauseSchedule(rdId);
+
+        if (schedule == null) {
+            return;
+        }
+
+        pause.pausedReason.value = schedule.StatusReason__c;
+        pause.scheduleId = schedule.Id;
+    }
+
+    /**
+     * @description Returns the Paused Reason value and label select options
+     * @return List<Option>
+     */
+    private static List<Option> getPausedReasonOptions() {
+        List<Option> options = new List<Option>();
+        List<Schema.PicklistEntry> entries = npe03__Recurring_Donation__c.ClosedReason__c.getDescribe().getPicklistValues();
+
+        for (Schema.PicklistEntry entry : entries) {
```

Comment:
I think this can be replaced by a call to `UTIL_Describe.getSelectOptions()`, or could this just be handled in JavaScript?

Reply:
The `UTIL_Describe.getSelectOptions()` contains a loop and since the `SelectOption` cannot be returned to the LWC (I am getting `Apex Type unsupported in JSON: System.SelectOption` error), so there should be another loop to convert options from the `SelectOption` to the (value, label) object list. The `getPicklistValues` can be used from the LWC, but then the page needs to check for one more Boolean if everything is loaded in addition to checking if has access, is closed, and is loading. The current implementation returns all Paused Reason data at once.
I am suggesting keeping the implementation as is unless there is a cleaner way to coordinate load and display of the Paused Reason.

Reply:
Understood. Makes me wonder if we eventually need a utility method for getPicklistValuesForLwc(object, field) that returns the wrapper class. Though even in that model there is no support for the _default_ value. That's probably not an issue here, but it's something to consider if we eventually want to move to a utility method.